bin/docker_operations.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+
+import docker
+import sys
+import toml
+import inspect
+import os
+
+self_directory = os.path.dirname(os.path.abspath(__file__))
+base_directory = os.path.join(self_directory, "..")
+version = toml.load(os.path.join(base_directory, "version.toml"))["version"]
+version_tag = "partto_{major}.{minor}.{fix}.{build}-{variant}".format(**version)
+volume_name = "partto_data"
+
+
+class Commander:
+    def __init__(self, client):
+        self.client = client
+
+    def ensure_volume(self):
+        try:
+            self.client.volumes.get(volume_name)
+            print("already have value")
+        except docker.errors.NotFound:
+            print("creating volume")
+            self.client.volumes.create(volume_name)
+            print("done creating volume")
+
+    def ensure_image(self):
+        try:
+            self.client.images.get(version_tag)
+        except docker.errors.ImageNotFound:
+            print("building image: this may take a while")
+            self.client.images.build(tag=version_tag, path=".")
```

Comment:
I'm thinking my version should be after a colon here, but I don't see it being tagged with anything here. Asking the internet.